### DISCUSSION TITLE
Examining memory with lldb

### DISCUSSION THREAD

**Discussion by ldelossa** [mfussenegger/nvim-dap#701] [Q&A]
Hello,

Is there a way to examine memory with nvim-dap or nvim-dap-ui ? 

I'm using lldb-vscode setup and in vscode I can get either look at a variable in a hex editor or get a repl with lldb to use the examine memory commands. 

Can I do either of this (preferably both) with nvim-dap and nvim-dap-ui?

**Comment by mfussenegger** [comment]
With lldb-vscode you can run any lldb commands in the dap-repl if you prefix them with ```. See https://github.com/mfussenegger/nvim-dap/wiki/Debug-Adapter-installation#lldb-commands

Other memory-view functionality is currently not implemented, but I'd appreciate hearing suggestions how that could look like from a UI/UX perspective.

**Reply by ldelossa** [reply]
Thank you @mfussenegger 

Re: Memory viewer

So, even in `vscode` it does not become a "first class citizen" instead, another plugin is required which handles the functionality. 

This plugins is very basic (which I think is good). It basically gives you memory addresses on the left, and hex values on the right. The only customization that can occur is specifying the number of bytes displayed per line. 

![image](https://user-images.githubusercontent.com/5642902/192355391-85bcf3c8-7c49-4d11-84c3-4a2de3009fe1.png)
Here is an example of examining memory in vscode. 

Anything that comes close to this UI I think is a great approach. 

IMO, this is a nice to have, just being able to run the lldb codes directly seem OK to me. But visuals are helpful!
